lib/tests/streamlit/config_util_test.py

```diff
@@ -146,3 +155,115 @@ def test_server_option_changed(self, old, new, changed):
+            "server.customOption": ConfigOption(
+                key="server.customOption",
+                description="""
+                    This option has multiple lines.
+                    Each line should be preserved.
+                    Even this one.
+                """,
+                default_val="default",
+                type_=str,
+            )
+        }
+
+        config_util.show_config(CONFIG_SECTION_DESCRIPTIONS, config_options)
+
+        [(args, _)] = patched_echo.call_args_list
+        # Remove the ascii escape sequences used to color terminal output.
+        output = re.compile(r"\x1b[^m]*m").sub("", args[0])
+        lines = set(output.split("\n"))
+
+        assert "# This option has multiple lines." in lines
+        assert "# Each line should be preserved." in lines
+        assert "# Even this one." in lines
+
+    @patch("streamlit.config_util.click.echo")
+    def test_omits_empty_lines_at_description_start(self, patched_echo):
+        config_options = {
+            "server.customOption": ConfigOption(
+                key="server.customOption",
+                description="""
+
+                    This option's description starts from third line.
+                    All preceding empty lines should be removed.
+                """,
+                default_val="default",
+                type_=str,
+            )
+        }
+
+        config_util.show_config(CONFIG_SECTION_DESCRIPTIONS, config_options)
+
+        [(args, _)] = patched_echo.call_args_list
+        # Remove the ascii escape sequences used to color terminal output.
+        output = re.compile(r"\x1b[^m]*m").sub("", args[0])
+        lines = set(output.split("\n"))
+
+        assert "# This option's description starts from third line." in lines
+        assert "# All preceding empty lines should be removed." in lines
+
+    @patch("streamlit.config_util.click.echo")
+    def test_description_appears_before_option(self, patched_echo):
+        config_options = {
+            "server.customOption": ConfigOption(
+                key="server.customOption",
+                description="This option's description should appear before the option.",
+                default_val="default",
+                type_=str,
+            )
+        }
+
+        config_util.show_config(CONFIG_SECTION_DESCRIPTIONS, config_options)
+
+        [(args, _)] = patched_echo.call_args_list
+        # Remove the ascii escape sequences used to color terminal output.
+        output = re.compile(r"\x1b[^m]*m").sub("", args[0])
+        lines = output.split("\n")
+
+        # Find the index of the description and the option in the output.
+        description_index = next(
+            i
+            for i, line in enumerate(lines)
+            if "This option's description should appear before the option." in line
+        )
+        option_index = next(
+            i for i, line in enumerate(lines) if 'customOption = "default"' in line
+        )
```

Comment:
Is it possible to use `lines.index` to do the same thing? I think it'd be cleaner unless the difficulty is that the exact text of the line (whitespace and all) is awkward to construct

Reply:
Yikes, should have thought of that! Way cleaner to use `lines.index`. Fixed on 533c1f8.